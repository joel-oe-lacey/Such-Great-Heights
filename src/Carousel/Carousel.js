import React, { Component } from 'react';
import './Carousel.scss';
import Arrow from './Arrow/Arrow';

export default class Carousel extends Component {
    constructor(props) {
        super(props)
        this.state = {
            imageIndex: 0,
        }
    }

    changeImg = direction => {
        let { imageIndex } = this.state;
        let newIndex;

        if (direction === 'left') {
            newIndex = imageIndex === 0 ? 2 : imageIndex - 1;
        } else {
            newIndex = imageIndex === 2 ? 0 : imageIndex + 1;
        }

        this.setState({
            imageIndex: newIndex
        });
    }

    render() {
        console.log(`../../images/${this.props.id}_${this.props.imgTags[this.state.imageIndex]}.jpg`)
        return <section className="carousel">
            <img className='images' src={`/../../images/${this.props.id}_${this.props.imgTags[this.state.imageIndex]}.jpg`} alt={this.props.name}/>
            <Arrow direction="left" changeImg={this.changeImg} />
            <Arrow direction="right" changeImg={this.changeImg} />
        </section>
    }
}
