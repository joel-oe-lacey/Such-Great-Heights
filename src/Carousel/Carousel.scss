.carousel {
    height: 100%;
    width: 100%;
    display: flex;
    flex-flow: row wrap;
}

.images {
  height: 100%;
  width: 100%;
  border-radius: 3%;

}

.arrow-btn {
    height: 10%;
    width: 45%;
    margin: 2%;
    border-radius: 4%;
    background-color: #FFD6BA;
    font-size: 1.5em;
    cursor: pointer;
}
