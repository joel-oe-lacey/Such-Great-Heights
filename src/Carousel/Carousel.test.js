import React from 'react';
import { shallow } from 'enzyme';
import Carousel from './Carousel';

describe('Carousel', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Carousel id={8} name={'test text'} imgTags={['a', 'b', 'c']} />);
    });

    it('Should equal snapshot', () => {
        const wrapper = shallow(<Carousel id={8} name={'test text'} imgTags={['a','b','c']} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('Should subtract image index for left click', () => {
        expect(wrapper.state('imageIndex')).toEqual(0);

        wrapper.instance().changeImg('left');

        expect(wrapper.state('imageIndex')).toEqual(3);
    });

    it('Should add to image index for right click', () => {
        expect(wrapper.state('imageIndex')).toEqual(0);

        wrapper.instance().changeImg('right');

        expect(wrapper.state('imageIndex')).toEqual(1);
    });
});