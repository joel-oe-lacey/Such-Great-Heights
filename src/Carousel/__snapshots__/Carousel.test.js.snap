// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Should equal snapshot 1`] = `
<section
  className="carousel"
>
  <img
    alt="test text"
    className="images"
    src="/../../images/8_a.jpg"
  />
  <Arrow
    changeImg={[Function]}
    direction="left"
  />
  <Arrow
    changeImg={[Function]}
    direction="right"
  />
</section>
`;
